### PR TITLE
[ICP-12456] Add preferences to Everspring Motion Sensor

### PR DESCRIPTION
**Pull request by MGoralczykS**
@greens @KKlimczukS @PKacprowiczS @ZWozniakS @MWierzbinskaS @BJanuszS 
Could you review this?

### CODE REVIEW

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 1]
```diff
@@ -61,18 +62,44 @@ metadata {
 		main "motion"
 		details(["motion", "battery", "tamper"])
 	}
+		preferences {
+			section{
+				input(tittle: "Everspring Motion Sensor settings",
```
Comment (cbaumler) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:67]: nit: "title"

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 2]
```diff
@@ -61,18 +62,44 @@ metadata {
...
+			}
+		}
 }
 
 def installed() {
 // Device wakes up every 4 hours, this interval allows us to miss one wakeup notification before marking offline
 	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
 	sendEvent(name: "tamper", value: "clear", displayed: false)
+	state.reTriggerMap = [:]
+	if(isEverspring()) {
```
Comment (cbaumler) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:90]: nit: space after `if`

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 3]
```diff
@@ -61,18 +62,44 @@ metadata {
...
+		state.reTriggerMap = reTriggerValue ? reTriggerValue : false
+	}
+	
 	response(initialPoll())
 }
 
 def updated() {
 // Device wakes up every 4 hours, this interval allows us to miss one wakeup notification before marking offline
 	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if(isEverspring()) {
```
Comment (cbaumler) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:100]: nit: space after `if`

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 4]
```diff
@@ -172,6 +199,11 @@ def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd)
 	} else {
 		result << response(zwave.wakeUpV1.wakeUpNoMoreInformation())
 	}
+	if(isEverspring()) {
```
Comment (cbaumler) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:202]: nit: space after `if`

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 5]
```diff
@@ -61,18 +62,44 @@ metadata {
 		main "motion"
 		details(["motion", "battery", "tamper"])
 	}
+		preferences {
+			section{
+				input(tittle: "Everspring Motion Sensor settings",
+					description: "The Configuration parameter that can be used to adjust the interval of being re-triggered after the detector has been triggered as Configuration Parameter #4.  No response will be made during this interval if a movement is presented.  The time interval can be set between 5 secs to 3600 secs.",
```
Comment (greens) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:68]: I'd simplify this text a little bit.

I'd recommend:

"Set the delay before the motion sensor can be re-triggered."

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 6]
```diff
@@ -61,18 +62,44 @@ metadata {
...
+	}
+	
 	response(initialPoll())
 }
 
 def updated() {
 // Device wakes up every 4 hours, this interval allows us to miss one wakeup notification before marking offline
 	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if(isEverspring()) {
+		state.reTriggerMap = reTriggerValue ? reTriggerValue : false
```
Comment (greens) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:101]: The values you're putting into this "map" are inconsistent. In addition to calling it a "map", here you're putting in either an integer or a boolean, while below you put a one-value array of `[0]`.

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy [hunk 7]
```diff
@@ -172,6 +199,11 @@ def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd)
 	} else {
 		result << response(zwave.wakeUpV1.wakeUpNoMoreInformation())
 	}
+	if(isEverspring()) {
+		result << response(zwave.configurationV1.configurationSet(parameterNumber: 4, size: 2, scaledConfigurationValue: reTriggerValue))
```
Comment (greens) [inline devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:203]: Shouldn't the decision to send this or not depend on the value in `reTriggerMap`? If not, why are you attempting to reset the value below?

File: devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy (hunk 6 above)
Comment (MGoralczykS) [reply devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:101]: So I should delete this map and leave only value on 203 line?

Comment (greens) [reply devicetypes/smartthings/zwave-motion-sensor.src/zwave-motion-sensor.groovy:101]: So your intention is to store a value when the setting is changed, send that value on the next device wakeup, and then clear it, and not send the setting on next wakeup if you've already sent it, right?

Here's what your code should be doing:
1. When the setting is changed, update `state.newTriggerValue`
2. When you receive a wakeup notification, check the value of `state.newTriggerValue`
3. If you have a non-`null` value, send the configuration command and pass in `state.newTriggerValue`
4. Set `state.newTriggerValue` to `null`

**Comment by k-morawski** [comment]
https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/32967

**Pull request closed without merging**